edx_exams/apps/core/models.py

```diff
@@ -234,6 +234,27 @@ def check_no_other_active_attempts_for_user(cls, user_id, attempt_id):
         except cls.DoesNotExist:
             return True
 
+    @classmethod
+    def get_attempt_by_attempt_number_and_resource_id(cls, attempt_number, resource_id):
```

Comment:
nothing wrong with having this here I just want to point out that we don't need always need this type of logic to be on the model class directly. If it's a pretty specific/narrow use case (filter by resource_id feels pretty narrow) it's totally fine to just put in on api.py. We don't always need the extra layer of abstraction.

My rule of thumb here is to do this logic on the model if I expect it would be used multiple times elsewhere.